src/amp-story-player/amp-story-player-impl.js

```diff
@@ -1264,6 +1263,39 @@ export class AmpStoryPlayer {
     );
   }
 
+  /**
+   * Returns the storyIdx of a story given a URL.
+   * @param {string} storyUrl
+   * @return {number}
+   * @private
+   */
+  getStoryIdxFromUrl_(storyUrl) {
```

Comment:
Given that we just refactored everything to only work with StoryDefs, this should return a StoryDef and not a number. If the caller happens to need the index, all it takes is `story.idx`